src/devtools/client/webconsole/components/Input/JSTerm.tsx

```diff
@@ -77,6 +76,13 @@ export default function JSTerm() {
       setHideAutocomplete(true);
     }
   };
+  // for use in e2e tests
+  const showAutocomplete = (show: boolean) => {
+    setHideAutocomplete(!show);
+    if (show) {
+      resetAutocompleteIndex();
+    }
+  };
```

Comment:
Having this here and then passed to the lower component just for the sake of it being able to attach itself to the `window` has so many things wrong with it that I can't even... 😅 

Maybe it's worth getting a ref to a `CodeMirror` (i.e. `WrappedCodeMirror`) and attaching to `window.jsterm` from within `JSTerm`, instead of `WrappedCodeMirror`.

If we could get some flag inside this impl (a la `isTest()`), so that it's rellay a passthrough for live code.

Reply:
> Having this here and then passed to the lower component just for the sake of it being able to attach itself to the window has so many things wrong with it that I can't even... :sweat_smile:

I agree totally, but refactoring this thing seemed out of the scope of this PR (and I didn't want to have 2 different ways to attach callbacks to `window.jsterm`, that would make the code even more confusing IMHO)

> If we could get some flag inside this impl (a la isTest()), so that it's rellay a passthrough for live code.

Done.